docs-src/concepts/what-is-an-activity.md

```diff
@@ -2,21 +2,21 @@
 id: what-is-an-activity
 title: What is an Activity?
 sidebar_label: Activity
-description: In day-to-day conversations, the term "Activity" frequently denotes either an Activity Type, an Activity Definition, or an Activity Execution.
+description: In day-to-day conversation, the term "Activity" denotes an Activity Type, Activity Definition, or Activity Execution.
 tags:
   - term
   - explanation
 ---
 
-In day-to-day conversations, the term _Activity_ frequently denotes either an [Activity Definition](/concepts/what-is-an-activity-definition), an [Activity Type](/concepts/what-is-an-activity-type), or an [Activity Execution](/concepts/what-is-an-activity-execution).
+In day-to-day conversation, the term _Activity_ denotes an [Activity Definition](/concepts/what-is-an-activity-definition), [Activity Type](/concepts/what-is-an-activity-type), or [Activity Execution](/concepts/what-is-an-activity-execution).
 Temporal documentation aims to be explicit and differentiate between them.
 
-An Activity is a normal function or object method that executes a single, well-defined action (either short or long running), such as calling another service, transcoding a media file, or sending an email message.
+An Activity is a normal function or method that executes a single, well-defined action (either short or long running), such as calling another service, transcoding a media file, or sending an email message.
+Activity code may be non-deterministic.
```

Comment:
I don't want to open a can of worms here, but might this be the place to talk about idempotence? Or at least mention it?

Reply:
```suggestion
Activity code may be non-deterministic.
We recommend that it be [idempotent](/activities#idempotency)
```

Reply:
Technically should be require, not recommend (unless retries are disabled)

Reply:
Yeah, that's what I meant by can of worms. Whatever has been agreed on is fine